sql-plugin/src/main/330+/scala/com/nvidia/spark/rapids/shims/GpuIntervalUtils.scala

```diff
@@ -315,15 +314,12 @@ object GpuIntervalUtils extends Arm {
    * @return micros column
    */
   private def getMicrosFromDecimal(sign: ColumnVector, decimal: ColumnVector): ColumnVector = {
-    withResource(decimal.castTo(DType.create(DType.DTypeEnum.DECIMAL64, -6))) { decimal =>
-      withResource(Scalar.fromLong(1000000L)) { million =>
-        withResource(decimal.mul(million)) { r =>
-          withResource(r.asLongs()) { l =>
-            l.mul(sign)
-          }
-        }
-      }
-    }
+    val timesMillion = lazyReduce(Seq(
```

Comment:
I do not think `lazyReduce`/`lazyResource` is buying us much in cases like this. In my opinion this new syntax is harder to read overall. I'd propose we still un-nest, but keep the `withResource` syntax:

```
val timesMillion = 
  withResource(decimal.castTo(DType.create(DType.DTypeEnum.DECIMAL64, -6))) { decimal =>
    withResource(Scalar.fromLong(1000000L)) { million =>
      decimal.mul(million)
    }
  }
```

Reply:
One of the main pieces of information that is now implicit with this is the reference counting, which is something that I believe we need to be really explicit about. The other piece is that it effectively makes use reverse polish notation when calling into cuDF. It feels we are inventing a new language, and this is why I find it hard to read (I have to decompile it).

Looking at the "chain of ORs" case (https://github.com/NVIDIA/spark-rapids/pull/6833/files#diff-e981882f5ee2f922528de849ae5397dd30e5bfe9dd5fdbe3421de4733a0eae1aR1224), or this other case (https://github.com/NVIDIA/spark-rapids/pull/6833/files#diff-4792f44f424103af4020d7560b14057217c13fa3a31ea2be3b20c426d5d550e5R184), I think it is harder to understand the lifetime of things, and also what is happening when we call the cuDF function.

I understand the reasoning behind it, where we can combine these lazy functions together, and it will execute in the most economical way. I'd hope we can do that without making things more complicated than they are right now.

Reply:
Note that's my opinion. I think it would be great to hear from others as to what the overall preference is.